### ISSUE TITLE
any plan to make it works with sonarqube?

### ISSUE THREAD

**Issue by junhui** [security-code-scan/security-code-scan#107]
any plan to make it works with sonarqube?

**Comment by JarLob** [comment]
No plans as I personally don't need it, but contributions with necessary changes are welcome.

**Comment by h3xstream** [comment]
Their was an official plugin that was deprecated (not very stable). The design of SonarQube does not allow easy integration with 3rd party analyzers.

One alternative is to use a CI that has the capability to report and track warning output from MsBuild.
https://wiki.jenkins.io/display/JENKINS/MSBuild+Plugin
This is far from a modern dashboard and reporting.

**Comment by projectrvce** [comment]
> No plans as I personally don't need it, but contributions with necessary changes are welcome.

What kind of changes are to be made in order to integrate it with sonarqube? Could you please help me in doing this

**Comment by projectrvce** [comment]
> Their was an official plugin that was deprecated (not very stable). The design of SonarQube does not allow easy integration with 3rd party analyzers.
> 
> One alternative is to use a CI that has the capability to report and track warnings output from MsBuild.
> https://wiki.jenkins.io/display/JENKINS/MSBuild+Plugin
> This is far from a modern dashboard and reporting.
will this help me in integrating the security code scan with sonarqube

**Comment by jvasallo** [comment]
I've opened a subsequent ticket on their side:
https://community.sonarsource.com/t/c-security-code-scan-sonarscanner/14829

Seeing as this has been implemented by Gitlab in their SAST tool, would be great to see if this can be integrated in Sonarqube as well. Similar to how the custom rules were implemented for Bandit in Python scans. Might just be setting up rule definitions on the Sonar side as well.

Will report back if we make any progress on this from our side.
